api/devicerepository.proto

```diff
@@ -0,0 +1,366 @@
+      string codec_id = 3 [(gogoproto.customname) = "CodecID"];
+    }
+    // Firmware version string.
+    string version = 1;
+    // Numeric firmware revision number.
+    uint32 numeric = 2;
+    // Hardware versions supported by this firmware version.
+    repeated string supported_hardware_versions = 3;
+    // Device profiles for each supported region (band).
+    map<string, Profile> profiles = 4;
+  }
+  // Available firmware versions.
+  repeated FirmwareVersion firmware_versions = 6;
+  // List of sensors included in the device.
+  repeated string sensors = 7;
+  message Dimensions {
+    // Device width (mm).
+    google.protobuf.FloatValue width = 1;
+    // Device height (mm).
+    google.protobuf.FloatValue height = 2;
+    // Device diameter (mm).
+    google.protobuf.FloatValue diameter = 3;
+    // Device length (mm).
+    google.protobuf.FloatValue length = 4;
+  }
+  // Device dimensions.
+  Dimensions dimensions = 8;
+  // Device weight (gram).
+  google.protobuf.FloatValue weight = 9;
+  message Battery {
+    // Whether the device battery can be replaced.
+    google.protobuf.BoolValue replaceable = 1;
+    // Battery type.
+    string type = 2;
+  }
+  // Device battery information.
+  Battery battery = 10;
+  message OperatingConditions {
+    message Limits {
+      // Min value of operating conditions range.
+      google.protobuf.FloatValue min = 1;
+      // Max value of operating conditions range.
+      google.protobuf.FloatValue max = 2;
+    }
+    // Temperature operating conditions (Celsius).
+    Limits temperature = 1;
+    // Relative humidity operating conditions (Fraction, in range [0, 1]).
+    Limits relative_humidity = 2;
+  }
+  // Device operating conditions.
+  OperatingConditions operating_conditions = 11;
+  // Device IP rating code.
+  string ip_code = 12 [(gogoproto.customname) = "IPCode"];
+  // Supported key provisioning methods.
+  repeated KeyProvisioning key_provisioning = 13;
+  // Device key security.
+  KeySecurity key_security = 14;
+
+  message Photos {
+    // Main device photo.
+    string main = 1;
+    // List of other device photos.
+    repeated string other = 2;
+  }
+  // Device photos.
+  Photos photos = 15;
+  message Videos {
+    // Link to main device video.
+    string main = 1;
+    // Links to other device videos.
+    repeated string other = 2;
+  }
+  // Device video.
+  string video = 16;
+  // Device videos.
+  Videos videos = 17;
+
+  // Device information page URL.
+  string product_url = 18 [(gogoproto.customname) = "ProductURL"];
+  // Device datasheet URL.
+  string datasheet_url = 19 [(gogoproto.customname) = "DatasheetURL"];
+
+  message Reseller {
+      // Reseller name.
+      string name = 1;
+      // Reseller regions.
+      repeated string region = 2;
+      // Reseller URL.
+      string url = 3 [(gogoproto.customname) = "URL"];
+  }
+  // Reseller URLs.
+  repeated Reseller resellers = 20 [(gogoproto.customname) = "Resellers"];
```

Comment:
It looks like this branch isn't correctly rebased. Please do so and re-request review. Also please check the files here before doing so, to verify that the diff is good for review.